content/en/blog/2024/otel-collector-anti-patterns/index.md

```diff
@@ -0,0 +1,176 @@
+---
+title: OpenTelemetry Collector Anti-Patterns
+linkTitle: OTel Collector Anti-Patterns
+date: 2024-02-23
+author: >-
+  [Adriana Villela](https://github.com/avillela) (Lightstep),
+
+canonical_url: https://adri-v.medium.com/43dca4a857a0
+---
+
+![House on stilts against ocean and mountain backdrop](house-on-stilts.jpg)
+
+The [OpenTelemetry Collector](/docs/collector) is one of my favorite
+OpenTelemetry (OTel) components. It’s a flexible and powerful data pipeline
+which allows you to ingest OTel data from one or more sources, transform it
+(including batching, filtering, and masking), and export it to one or more
+observability backends for analysis. It’s vendor-neutral. It’s extensible,
+meaning that you can create your own custom components for it. What’s there not
+to like?
+
+And like many tools out there, it is also very easy to fall into some bad
+habits. Today, I will dig into five OpenTelemetry Collector anti-patterns, and
+how to avoid them. Let’s get started!
+
+## Anti-Patterns
+
+### 1- Not using the OpenTelemetry Collector
```

Comment:
I'm not sure how I feel about this one: there are certainly use-cases where people don't want or can't use a collector. For instance, people collecting data at the edge from IOT devices might have a better time sending data to their backend instead of a local collector, given that resources on that edge might be limited.

I would certainly state that having a Collector is a great general pattern, but not having it isn't necessarily an anti-pattern IMO.

Reply:
I think it's great to use this one for attention grabbing. I agree that it'd need some explanation after the heading following @jpkrohling 's arguments.

Reply:
The problem is that more people read the titles than the whole text, and that's the message they'll get and pass forward. Perhaps the title could be "1 - Not using the OTel Collector (with exceptions)" or something like that.